neuraxle/pipeline.py

```diff
@@ -158,52 +149,31 @@ def load(self, pipeline: 'Pipeline', data_container: DataContainer) -> Optional[
         :return: pipeline
         """
         pipeline_cache_folder = os.path.join(self.cache_folder, pipeline.name)
-        pipeline_cache_list_file_name_path = os.path.join(pipeline_cache_folder, self.pipeline_cache_list_file_name)
-
-        with open(pipeline_cache_list_file_name_path, mode='r') as file:
-            lines = file.readlines()
 
-            pipeline_path_line_index = self._find_saved_pipeline_path_line_index(
-                data_container=data_container,
-                lines=lines
+        return load(
+            self._create_next_cached_pipeline_path(
+                pipeline_cache_folder,
+                data_container
             )
-            saved_pipeline_path = lines[pipeline_path_line_index].strip()
-
-            return load(saved_pipeline_path)
-
-    @staticmethod
-    def _find_saved_pipeline_path_line_index(data_container, lines):
-        """
-        Find saved pipeline path line
-
-        :param data_container:
-        :param lines:
-        :return:
-        """
-        for index, line in enumerate(lines):
-            if str(data_container) == line.strip():
-                return index + 1
-        return -1
+        )
 
-    def _create_next_cached_pipeline_path(self, pipeline_cache_folder) -> str:
+    def _create_next_cached_pipeline_path(self, pipeline_cache_folder, data_container: 'DataContainer') -> str:
```

Comment:
what about `def _step_path_from_cache_folder_and_hash`?

Reply:
you are commenting commits from another pr. resolving here.